src/interfaces/IBPool.sol

```diff
@@ -54,53 +131,141 @@ interface IBPool is IERC20 {
     address tokenOut,
     uint256 tokenAmountOut,
     uint256 maxPoolAmountIn
   ) external returns (uint256 poolAmountIn);
 
+  /**
+   * @notice Gets the spot price of a hypothetical swap
+   * @param tokenIn The address of the token to swap in
+   * @param tokenOut The address of the token to swap out
+   * @return spotPrice The spot price of the swap
+   */
   function getSpotPrice(address tokenIn, address tokenOut) external view returns (uint256 spotPrice);
 
+  /**
+   * @notice Gets the spot price of a hypothetical swap without the fee
+   * @param tokenIn The address of the token to swap in
+   * @param tokenOut The address of the token to swap out
+   * @return spotPrice The spot price of the swap without the fee
+   */
   function getSpotPriceSansFee(address tokenIn, address tokenOut) external view returns (uint256 spotPrice);
 
+  /**
+   * @notice Gets the finalized status of the pool
+   * @return isFinalized True if the pool is finalized, False otherwise
+   */
   function isFinalized() external view returns (bool isFinalized);
 
+  /**
+   * @notice Gets the bound status of a token
+   * @param t The address of the token to check
+   * @return isBound True if the token is bound, False otherwise
+   */
   function isBound(address t) external view returns (bool isBound);
 
+  /**
+   * @notice Gets the number of tokens in the pool
+   * @return numTokens The number of tokens in the pool
+   */
   function getNumTokens() external view returns (uint256 numTokens);
 
+  /**
+   * @notice Gets the current array of tokens in the pool, while the pool is not finalized
+   * @return tokens The array of tokens in the pool
+   */
   function getCurrentTokens() external view returns (address[] memory tokens);
 
+  /**
+   * @notice Gets the final array of tokens in the pool, after finalization
+   * @return tokens The array of tokens in the pool
+   */
   function getFinalTokens() external view returns (address[] memory tokens);
 
+  /**
+   * @notice Gets the denormalized weight of a token in the pool
+   * @param token The address of the token to check
+   * @return denormWeight The denormalized weight of the token in the pool
+   */
   function getDenormalizedWeight(address token) external view returns (uint256 denormWeight);
 
+  /**
+   * @notice Gets the total denormalized weight of the pool
+   * @return totalDenormWeight The total denormalized weight of the pool
+   */
   function getTotalDenormalizedWeight() external view returns (uint256 totalDenormWeight);
 
+  /**
+   * @notice Gets the normalized weight of a token in the pool
+   * @param token The address of the token to check
+   * @return normWeight The normalized weight of the token in the pool
+   */
   function getNormalizedWeight(address token) external view returns (uint256 normWeight);
 
+  /**
+   * @notice Gets the balance of a token in the pool
+   * @param token The address of the token to check
+   * @return balance The balance of the token in the pool
```

Comment:
hmmm "in the pool" for balances sounds wrong, i'd say "Pool's ERC20 balance" or sth like that, but the balance is never IN the pool, the pool holds the balance